text/typeclass_control.md

```diff
@@ -0,0 +1,19 @@
+- Title: Typeclasses
+
+- Drivers: Cyril Cohen
+
+- Status: Draft
+
+----
+
+# Wishes
+
+- A cut tag between the (class) arguments of instances, with the same behavior as the prolog cut (as opposed to the Hint Cut). Syntax TBD
+E.g. (using "cut" as a placeholder for the future syntax)
+```coq
+Instance A `{C x} "cut" `{C' x y} : C (f x y) := ...
+```
+If a solution for (C x) is found, then no backtrack is allowed if a search for (C' x y) fails.
+
+- Control typeclass opacity by class and not globally. The opacity/transparency constraint should be stored in a hint database in order for it to parametrize typeclasses eauto with ...
+
```

Comment:
One way to do this would be to have one database per class, with the option of putting instances in multiple databases, is that your next wish? We could also associate a transparency state to each class of course (to each head in the hint database).

Reply:
@mattam82 any progress on that?